geonode_mapstore_client/client/js/components/DetailsPanel/DetailsAttributeTable.jsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2023, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React, { useState, useEffect } from 'react';
+import { FormattedMessage } from 'react-intl';
+import PropTypes from 'prop-types';
+
+import {
+    getDatasetByPk
+} from '@js/api/geonode/v2'
+
+import Table from '@js/components/Table';
+
+
+const parseAttributeData = (dataset) => {
+    if (dataset?.attribute_set) {
+        const header = [{
+            value: "Name",
+            key: "name"
+        }, {
+            value: "Label",
+            key: "label"
+        }, {
+            value: "Description",
+            key: "description"
+        }]
+
+        const na = <FormattedMessage id="gnhome.na" defaultMessage="N/A" />
+        const rows = dataset.attribute_set.map(attribute => ({
+            name: attribute.attribute,
+            label: attribute.attribute_label || na,
+            description: attribute.description || na,
+        }));
+
+        return { header, rows };
+    }
+
+    return { header: [], rows: [] };
+};
+
+
+const DetailsAttributeTable = ({
+    resource,
+}) => {
+    const [attributeData, setAttributeData] = useState({ header: [], rows: [] });
+    useEffect(() => {
+        const getAttributes = async () => {
+            if (resource.resource_type === "dataset") {
+                const dataset = await getDatasetByPk(resource.pk);
+                setAttributeData(parseAttributeData(dataset));
+            }
+        }
+        getAttributes();
+    }, [])
```

Comment:
The code in general looks good but I was reviewing the workflow on how we populate the detail panel data and I noticed that inside the dataset viewer the `attribute_set` was already available in the response while in the catalogue not. The main reason is the different epic used by the two components:

- the catalogue uses [getResourceByPk](https://github.com/GeoNode/geonode-mapstore-client/blob/master/geonode_mapstore_client/client/js/epics/gnsearch.js#L290) for all resource_type but the document
- the dataset viewer uses [getDatasetByPk](https://github.com/GeoNode/geonode-mapstore-client/blob/master/geonode_mapstore_client/client/js/epics/gnresource.js#L105) correctly

My proposal is to remove this internal request inside the component and update the [detail catalogue epic](https://github.com/GeoNode/geonode-mapstore-client/blob/master/geonode_mapstore_client/client/js/epics/gnsearch.js#L290) to use `getDatasetByPk` for dataset to avoid completely this internal request that it will be a duplication.

Reply:
good catch .. makes the extra query unneccessary :+1: